server/file_hooks.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+// FileWillBeUploaded is invoked after a file is uploaded.
+//
+// This demo implementation logs a message to the demo channel in the team
+// when a new file is uploaded.
+func (p *Plugin) FileWillBeUploaded(c *plugin.Context, fileInfo *model.FileInfo, reader bytes.Reader, buf *bytes.Buffer) {
```

Comment:
You can do fancy stuff like changing a file or rejecting it based on the content. Not blocking, but it would be cool to have this in the demo plugin.

Reply:
Added a check for size.